src/azul/terra.py

```diff
@@ -278,6 +284,30 @@ def insufficient_access(self, resource: str):
         )
 
 
+class TerraClientException(Exception):
+    pass
+
+
+class TerraTimeoutException(TerraClientException):
+
+    def __init__(self, url: str, timeout: float):
+        super().__init__(f'No response from {url!r} within {timeout} seconds')
+
+
+class TerraStatusException(TerraClientException):
+
+    def __init__(self, url: str, response: HTTPResponse):
+        super().__init__(f'Unexpected response from {url!r}',
+                         response.status, response.data)
+
+
+class TerraNameConflictException(TerraClientException):
+
+    def __int__(self, url: str, source_name: str, response: HTTPResponse):
+        super().__init__(f'More than one source named {source_name!r}',
+                         url, response)
```

Comment:
Didn't you determine that the `repr` of HTTPResponse is not very useful? If so, why are you using it here? Or is the type hint wrong?

Reply:
Type hint was wrong